README.md

```diff
@@ -82,141 +143,167 @@ assign res = '{}' | parse_json | hash_merge: redirect_to: redirect_to
+If you receive 500 error after modifying the permissions.liquid file, please check logs for the hint - most likely after the modifications it is no longer a valid JSON due to missing comma, an extra comma, or something similar. Typical errors are leaving a traililng comma. Here is an example of a modification that will raise an error:
 
 ```
-function profile = 'lib/quieries/profiles/update', user_id: params.updated_user.id, first_name: params.hook_params.first_name
-assign result = '{}' | parse_json | hash_merge: profile: profile
-return result
+  "foo": [
+    "foo.show",
+    "site.manage", <- this comma will cause an error
+  ]
 ```
 
-#### hook_user_delete
+## Customizing the module
+
+You can easily customize the module using [overwriting a module file](https://documentation.platformos.com/developer-guide/modules/modules#overwritting-a-module-file). In this section you will find some most common examples.
+
+### Example 1: Changing sign up URL
+
+By default the module creates [Page](https://documentation.platformos.com/developer-guide/pages/pages) which defines `/users/new` endpoint to display the registration form. If you would like to change it, for example to `/sign-up`, you can do it by creating a file overwrite for this page and manually defining `slug: sign-up`. 
 
-Fires when the user is deleted. The deleted user is added to `params.user`.
+First, create nested directories to be able to place your file overwrite there:
 
+`mkdir -p app/modules/user/public/views/pages/users/`
+
+Copy the page you would like to overwrite:
+
+`cp modules/user/public/views/pages/users/new.liquid app/modules/user/public/views/pages/users/new.liquid`
+
+Edit the overwrite `app/modules/user/public/views/pages/users/new.liquid` YAML section of the Page (make sure you are modifying the overwrite and not the original file - overwrites are inside `<project root>/app/modules` directory, whereas original files will be in `<project root>/modules`):
+
+```yaml
+---
+slug: sign-up
+---
 ```
-function _delete_ = 'modules/user/commands/profiles/delete', user_id: params.user.id
-return nil
+```liquid
+{% raw %}
+{% liquid
+  function current_user = 'modules/user/queries/user/current'
+  include 'modules/user/helpers/can_do_or_unauthorized', requester: current_user, do: 'users.register'
+
+  function registration_fields = 'modules/user/queries/registration_fields/load'
+  theme_render_rc 'modules/user/users/new', context: context, registration_fields: registration_fields
+%}
+{% endraw %}
 ```
 
-#### hook_user_login
+As the end result, the registration form will be located now in /sign-up, and /users/new will render 404 Not Found error.
+
+### Example 2: Modifying the HTML of the sign in form
+
+First, create the directory for the overwrite. Because we want to override the presentation layer, we should overwrite the [Partial](https://documentation.platformos.com/developer-guide/pages/reusing-code-across-multiple-pages). 
```

Comment:
```suggestion
Steps:
1. Create the directory for the overwrite. Since we are overriding the presentation layer, we should overwrite the [Partial](https://documentation.platformos.com/developer-guide/pages/reusing-code-across-multiple-pages).
```